src/dialog-user/services/dialogData.ts

```diff
@@ -48,8 +46,26 @@ export default class DialogDataService {
    *
    **/
   private updateFieldSortOrder(data) {
-    let values = _.sortBy(data.values, data.options.sort_by === 'value' ? 0 : 1);
-    return data.options.sort_order === 'ascending' ? values : values.reverse();
+    const SORT_DESCRIPTION = 1;
+    const SORT_VALUE = 0;
+    const FIRST_OPTION = 0;
+    const VALUE = 0;
+    const sortBy = (data.options.sort_by === 'value' ? SORT_VALUE : SORT_DESCRIPTION);
+    let tempValues = data.values;
+    let defaultDropdownField = [];
+    // The following if deals with a empty default option if it exists
+    if (data.data_type === 'integer' && _.isNaN(tempValues[FIRST_OPTION][VALUE])) {
+      defaultDropdownField = tempValues.shift();
+    } else if(_.isNull(tempValues[FIRST_OPTION][VALUE])) {
+      defaultDropdownField = tempValues.shift();
+    }
```

Comment:
Should we just combine this `else if` and the `if` with an `||` since both lines 58 and 60 do the same thing?

Reply:
I thought about that when i reviewed my own code.  I think I will try that but will put the or on a separate line